README.adoc

```diff
@@ -0,0 +1,116 @@
+
+[source,bash]
+----
+$ git clone https://github.com/arquillian/ike-prow-plugins $GOPATH/src/github.com/arquillian/ike-prow-plugins
+----
+
+NOTE: Have a look how link:https://github.com/moovweb/gvm[Go Version Manager] can help you simplifying configuration and
+management of different versions of Go.
+
+NOTE: You can also use `setup.sh` which takes care of all of the above
+
+==== GitHub settings [[gh-settings]]
+
+You will need two secrets to be able to integrate with GitHub. The `hmac.token` file should contain the token that
+you give to GitHub for validating webhooks. You can generate it using any reasonable randomness-generator, for example
+link:http://random.org[random.org].
+
+The `oauth.token` is an OAuth2 token that has read and write access to the GitHub account. Generate it from
+link:https://github.com/settings/tokens/new[here].
+
+NOTE: Both of these files are ignored by git (see `.gitignore`) so you can keep them in your repository, as some `make`
+targets rely on them.
+
+=== Local Setup
+
+==== Minishift
+
+You can use Minishift for developing plugins locally.
+Follow link:https://docs.openshift.org/latest/minishift/getting-started/installing.html#installing-instructions[official installation guide] to set it up.
+
+===== Initial setup
+
+Before you start developing new plugins or evolving existing ones you need to deploy set of basic Prow services and configuration.
+
+For this purpose use `oc-init` target which will:
+
+* Create new project named `ike-prow-plugins`
+* Add config maps for Prow configuration (`config.yaml`) and its plugins (`plugins.yaml`)
+* Add required secrets (see <<gh-settings>> for details)
+* Apply `cluster/starter.yaml` with basic Prow infra.
+
+The key piece of this deployment is the `hook` service which listens to GitHub events and dispatches them to relevant plugins.
+
+You will have to create a route for this service. Go to `Applications -> Services -> hook` and click `Create Route`.
+Set `/hook` as a path and click `Create`.
+
+===== Pushing to local Minishift repository
+
+To speed up development cycle (and avoid pushing work-in-progress) images to Docker Hub you can use built-in docker repository available in Minishift.
+`Makefile` and templates are adjusted to use it, so the only thing you have to do is to set up few environment variables.
+
+[source,bash]
+----
+eval $(minishift docker-env)
+docker login -u $(oc whoami) -p $(oc whoami -t) $(minishift openshift registry)
+export REGISTRY=$(minishift openshift registry)
+export DOCKER_REPO=$(oc get project --show-all=false -o name | cut -d'/' -f 2)
+----
+
+IMPORTANT: Before executing `docker login` make sure you are logged in using token, as that's what is required returned
+by `oc whoami -t`. To do that open OpenShift Console (e.g. `xdg-open https://$(minishift ip):8443/console &>/dev/null'`),
+click on your username in the upper-right corner and then `Copy Login Command`. Use it to login from the terminal and
```

Comment:
I'm not sure if it should be mentioned somewhere that the user should use some of the latest versions of minishift as in the older versions there is no such an option.

Reply:
Will add to prerequisites. Good spot.